core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -189,5 +227,101 @@ public CategoryConfiguration forCategoryType(@NonNull CategoryType categoryType)
             return categoryConfiguration;
         }
     }
+
+    /**
+     * Loads a builder for an {@link AmplifyConfiguration} from an amplifyconfiguration.json file.
+     *
+     * @param context Context needed for reading JSON file
+     * @return An Amplify configuration builder instance
+     * @throws AmplifyException If there is a problem in the config file
+     */
+    @NonNull
+    public static Builder loadConfigFile(@NonNull Context context) throws AmplifyException {
```

Comment:
Maybe consider moving the `loadConfigFile` method into the `Builder` and create a builder method for `AmplifyConfiguration` that creates an instance of that `Builder`. So the callers would do something like this:

```
AmplifyConfiguration
    .builder() 
    .fromConfigFile(context)
    .forCategoryType(<category type>)
    .build();
```

It would be a bit more consistent with how other builders are structured throughout the framework.

Reply:
The reason for this implementation approach is that we want user to choose one specific way to load in details for the builder. Otherwise, a user may do 

```
AmplifyConfiguration.builder()
    .fromConfigFile(context)
    .fromJson(json) // what does this do?
    .fromConfigFile(anotherContext) // ????
    .build();
```

My implementation guarantees that one and only one loading approach is used for building a configuration.